hatchery/gen3licenseusermaps.go

```diff
@@ -0,0 +1,369 @@
+	if licenseInfo.FilePath == "" {
+		fmt.Printf("Error in container config. Empty FilePath for container %s\n", containerName)
+		ok = false
+	}
+	if licenseInfo.WorkspaceFlavor == "" {
+		fmt.Printf("Error in container config. Empty WorkspaceFlavor for container %s\n", containerName)
+		ok = false
+	}
+	return ok
+}
+
+var getActiveGen3LicenseUserMaps = func(dbconfig *DbConfig, container Container) (gen3LicenseUserMaps *[]Gen3LicenseUserMap, err error) {
+	// Query the table to get all active gen3 license user map items
+
+	targetEnvironment := os.Getenv("GEN3_ENDPOINT")
+	validateContainerLicenseInfo(container.Name, container.License)
+
+	// Query on global secondary index and filter by license type (eg. "STATA")
+	keyEx1 := expression.Key("environment").Equal(expression.Value(aws.String(targetEnvironment)))
+	keyEx2 := expression.Key("isActive").Equal(expression.Value("True"))
+	filt := expression.Name("licenseType").Equal(expression.Value(container.License.LicenseType))
+	expr, err := expression.NewBuilder().WithKeyCondition(expression.KeyAnd(keyEx1, keyEx2)).WithFilter(filt).Build()
+	if err != nil {
+		Config.Logger.Printf("Error in building expression for query: %s", err)
+		return nil, err
+	}
+	res, err := dbconfig.DynamoDb.Query(&dynamodb.QueryInput{
+		TableName:                 aws.String(Config.Config.LicenseUserMapsTable),
+		IndexName:                 aws.String(Config.Config.LicenseUserMapsGSI),
+		ExpressionAttributeNames:  expr.Names(),
+		ExpressionAttributeValues: expr.Values(),
+		KeyConditionExpression:    expr.KeyCondition(),
+		FilterExpression:          expr.Filter(),
+	})
+	if err != nil {
+		Config.Logger.Printf("Error in active user query: %s", err)
+		return nil, err
+	}
+
+	// Populate list of all active gen3 license user maps
+	var gen3LicenseUsers []Gen3LicenseUserMap
+	err = dynamodbattribute.UnmarshalListOfMaps(res.Items, &gen3LicenseUsers)
+	if err != nil {
+		Config.Logger.Printf("Error in unmarshalling active gen3 license user maps: %s", err)
+		return nil, err
+	}
+
+	Config.Logger.Printf("Debug: active gen3 license user maps %v", gen3LicenseUsers)
+	return &gen3LicenseUsers, nil
+}
+
+var getLicenseUserMapsForUser = func(dbconfig *DbConfig, userId string) (gen3LicenseUserMaps *[]Gen3LicenseUserMap, err error) {
+	// Query the table to get all gen3 license user map items for user
+
+	targetEnvironment := os.Getenv("GEN3_ENDPOINT")
+
+	// Query on global secondary index and filter by userId
+	keyEx1 := expression.Key("environment").Equal(expression.Value(aws.String(targetEnvironment)))
+	keyEx2 := expression.Key("isActive").Equal(expression.Value("True"))
+	filt := expression.Name("userId").Equal(expression.Value(userId))
+	expr, err := expression.NewBuilder().WithKeyCondition(expression.KeyAnd(keyEx1, keyEx2)).WithFilter(filt).Build()
+	if err != nil {
+		Config.Logger.Printf("Error in building expression for query: %s", err)
+		return nil, err
+	}
+	res, err := dbconfig.DynamoDb.Query(&dynamodb.QueryInput{
```

Comment:
same here. if might get long so maybe the query and loop can be moved to a function that `getActiveGen3LicenseUserMaps` and `getLicenseUserMapsForUser` can share

Reply:
Yes, good idea! Both `get` functions now use the same function for the query, `getItemsFromQuery`.